Doc/library/io.rst

```diff
@@ -1080,15 +1087,50 @@ Text I/O
       # .getvalue() will now raise an exception.
       output.close()
 
+      # docoding newlines for universal newlines mode.
+      io.StringIO('1\r\n2\r3\n4', newline=None).getvalue()
+      # '1\n2\n3\n4'
+
 
 .. index::
    single: universal newlines; io.IncrementalNewlineDecoder class
 
-.. class:: IncrementalNewlineDecoder
+.. class:: IncrementalNewlineDecoder(decoder, translate, errors='strict')
 
    A helper codec that decodes newlines for :term:`universal newlines` mode.
-   It inherits :class:`codecs.IncrementalDecoder`.
+   It inherits :class:`codecs.IncrementalDecoder`. It also records the types
+   of newlines encountered. When used with ``translate=False``, it ensures
+   that the newline sequence is returned in one piece. If an incremental
+   decoder is not required, :class:`TextIOWrapper` or :class:`StringIO`
+   should be used.
+
+   *decoder* is an incremental decoder or ``None``. If set to ``None`` it
+   decodes strings.
+
+   If *translate* is set to true, translate ``'\r\n'`` and ``'\r'`` into
+   ``'\n'``.
+
+   The *errors* argument works like that of
+   :class:`codecs.IncrementalDecoder`.
+
+   .. attribute:: newlines
+
+      a tuple of types of newlines encountered, or the newline if only
+      one type of newline is found.
```

Comment:
The types are not clear here.  Are you saying this is a tuple of strings or one string?

Reply:
the type is either a string or tuple of strings
https://github.com/python/cpython/blob/e18d81569fa0564f3bc7bcfd2fce26ec91ba0a6e/Lib/_pyio.py#L1971-L1980

Reply:
ok!  I wouldn’t say _types of newlines_, unless that phrase is used elsewhere in the docs, it could be confusing.

Reply:
how about changing to _representations of newlines_?
https://github.com/python/cpython/blob/e18d81569fa0564f3bc7bcfd2fce26ec91ba0a6e/Lib/_pyio.py#L1900-L1901

Reply:
That sounds a bit abstract to me.  These are newlines, meaning strings.
Isn’t there an existing phrase in io docs or dosctrings that can be reused?

Reply:
I can't find any in the `io` docs, The docstring of that class says _types of newlines_.

Reply:
It might be worth just giving some examples.

Reply:
Listed all the possible values.